test/socket.jl

```diff
@@ -1,178 +1,190 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-@test ip"127.0.0.1" == IPv4(127,0,0,1)
-@test ip"192.0" == IPv4(192,0,0,0)
-
-# These used to work, but are now disallowed. Check that they error
-@test_throws ArgumentError parse(IPv4, "192.0xFFF") # IPv4(192,0,15,255)
-@test_throws ArgumentError parse(IPv4, "192.0xFFFF") # IPv4(192,0,255,255)
-@test_throws ArgumentError parse(IPv4, "192.0xFFFFF") # IPv4(192,15,255,255)
-@test_throws ArgumentError parse(IPv4, "192.0xFFFFFF") # IPv4(192,255,255,255)
-@test_throws ArgumentError parse(IPv4, "022.0.0.1") # IPv4(18,0,0,1)
-
-@test UInt(IPv4(0x01020304)) == 0x01020304
-@test Int(IPv4("1.2.3.4")) == Int(0x01020304) == Int32(0x01020304)
-@test Int128(IPv6("2001:1::2")) == 42540488241204005274814694018844196866
-@test_throws InexactError Int16(IPv4("1.2.3.4"))
-@test_throws InexactError Int64(IPv6("2001:1::2"))
-
-let ipv = parse(IPAddr, "127.0.0.1")
-    @test isa(ipv, IPv4)
-    @test ipv == ip"127.0.0.1"
-end
+@testset "parsing" begin
+    @test ip"127.0.0.1" == IPv4(127,0,0,1)
+    @test ip"192.0" == IPv4(192,0,0,0)
```

Comment:
Is there a reference for this behavior? I found it kind of surprising and I'm investigating a few related IPv4 parsing oddities right now. My expectation would be to throw `ArgumentError` for this case and any other with less than 4 octets.

Reply:
Just did a bit more digging, and it looks like this is intentionally matching traditional `inet_aton` behavior, which allows the final number to be "the rest of the bits". E.g.

```julia
julia> map(Int, [0x01020304, 0x020304, 0x0304, 0x04])
4-element Array{Int64,1}:
 16909060
   131844
      772
        4

julia> [ip"16909060", ip"1.131844", ip"1.2.772", ip"1.2.3.4"]
4-element Array{Sockets.IPv4,1}:
 ip"1.2.3.4"
 ip"1.2.3.4"
 ip"1.2.3.4"
 ip"1.2.3.4"
```

So I guess there's no issue.